Libraries/LibWeb/IndexedDB/Internal/Algorithms.cpp

```diff
@@ -2051,4 +2051,31 @@ void queue_a_database_task(GC::Ref<GC::Function<void()>> steps)
     HTML::queue_a_task(HTML::Task::Source::DatabaseAccess, nullptr, nullptr, steps);
 }
 
+// https://w3c.github.io/IndexedDB/#cleanup-indexed-database-transactions
+bool cleanup_indexed_database_transactions(GC::Ref<HTML::EventLoop> event_loop)
+{
+    IGNORE_USE_IN_ESCAPING_LAMBDA bool has_matching_event_loop = false;
```

Comment:
I believe the use of escaping here is needed because we are using a GC::Function, but I don't believe using GC::Function is neccessary here since this is all syncronous and everything should be owned by the caller. So if `Database::for_each_database` took a standard Function or was a template callback, I believe this attribute could be removed